### DISCUSSION TITLE
Persisted data gets duplicated when used with react-native-mmkv

### DISCUSSION THREAD

**Discussion by franferns** [TanStack/query#4701] [Q&A]
Hi, 

I am trying to get the react query cache persisted using react-native-mmkv. Please find link with the code.

[Code with react-native-mmkv persister](https://snack.expo.dev/@francis25/sandbox-react-native?platform=android)

The problem with this is i think with everytime there is a save the data is appended to the file where the data is persisted

Here is how the persisted file looks like

![image](https://user-images.githubusercontent.com/779253/209428053-a556e8ae-ec76-450c-ac3b-7e78c9595a74.png)

Any idea as to what i am missing to get this to working ? 

Thanks & Regards,
Francis P.

**Comment by TkDodo** [comment]
> The problem with this is i think with everytime there is a save the data is appended to the file where the data is persisted

I can't get the example to work sadly. Maybe you can add some logging here:

```
persistClient: client => {
  storage.set(idbValidKey, JSON.stringify(client));
},
```

what is client, and what is the stringified version of it. If that is what you'd expect it to be, the issue would be that `storage.set` would append the data rather than replace it.

**Reply by franferns** [reply]
Thanks for your reply. The client remains the same for sometime but after cache increases to around 35-40 MB the saved data increases drastically.

Below is a video of how the issue looks on phone
[Reproducible video of storage increasing in size](https://drive.google.com/file/d/1v9eZLjDqmVDTmaMHz5Yem-YNZ3ECJp1u/view)

Please observe the **storage** section of the app 
1. Initially the **User data** and the **Cache** is zero (First time the app is installed not yet opened)
2. The app is opened and it makes a call to the rest api. The rest api takes sometime and returns around 35 MB of data. The data returned is a list of files converted to base64 format . The api takes some time to resolve data.
3. On opening of the **App info** for the app and the **Storage** observe the **Cache** every time its refreshed the **cache size** is increased it increases until it reaches around 35-40 MB after which the User data size drastically increases 500 MB +.

Please also find attached minimum code react native project 
[Minimal Code Project With the issue](https://drive.google.com/file/d/1-qfIanecqpsckUMcr7GZxa18mrw1QbaC/view?usp=sharing)

Thanks and Regards,
Francis P.

**Comment by franferns** [comment]
The problem I think is with the api returning way to much of base64 data . The better solution would be to change the rest api to return only the file path and download the files separately so its not a part of the persisted data. 

Thanks for all the time spent on this : )
